src/metabase/api/public.clj

```diff
@@ -190,17 +190,15 @@
 (defn public-dashcard-results-async
   "Return the results of running a query with `parameters` for Card with `card-id` belonging to Dashboard with
   `dashboard-id`. Throws a 404 immediately if the Card isn't part of the Dashboard. Returns a `StreamingResponse`."
-  [dashboard-id card-id export-format parameters & {:keys [qp-runner]
-                                                    :or   {qp-runner qp/process-query-and-save-execution!}
-                                                    :as   options}]
+  {:arglists '([& {:keys [dashboard-id card-id dashcard-id export-format parameters] :as options}])}
```

Comment:
Maybe even a docstring or a simple schema type `{:dashboard-id s/Int card-id s/Int dashcard-id s/Int}` although I'm not sure how well that plays with the varargs signature here. Would we have to do a manual schema check ourselves instead of `s/defn`? Again, not the end of the world and this stuff doesn't change much, but forgetting the dashcard-id could be an annoying bug in the future.

Reply:
My main goal here was to avoid having so many positional arguments which also gets easy to mix up, especially when we convert them to keyword arguments anyway.

This actually does get validated, just not here. It just passes off to `metabase.query-processor.dashboard/run-query-for-dashcard-async` which ends up validating those parameters [here](https://github.com/metabase/metabase/blob/fix-ambiguous-run-query-for-dashcard-endpoints/src/metabase/query_processor/dashboard.clj#L96-L104).

I agree tho that it's probably worth a mention in the docstring

Reply:
I went ahead and updated the docstring to make it clear that these parameters are required. I think that should be sufficient since you'll still get a schema validation error two or three frames down the line that `dashcard-id` is required if you don't pass it

Reply:
Perfect